src/hotspot/share/gc/g1/g1FullCollector.hpp

```diff
@@ -55,6 +57,15 @@ class G1FullGCSubjectToDiscoveryClosure: public BoolObjectClosure {
   }
 };
 
+// Full GC Mark that holds GC id and CPU time trace. Needs to be separate
+// from the G1FullCollector and G1FullGCScope to get consistent logging.
```

Comment:
I think saying just "consistent logging" here is too terse; some elaboration (background info and/or with/without this separation comparison) could make the reasoning more explicit.

Reply:
```suggestion
// from the G1FullCollector and G1FullGCScope to allow the Full GC logging
// to have the same structure as the Young GC logging.
```

Would this be enough? I don't want to add to many details as it might easily get stale. But I wanted the comment to let the reader know why this isn't folded in to the other structures.